pinot-core/src/main/java/org/apache/pinot/core/plan/GlobalPlanImplV0.java

```diff
@@ -49,6 +49,9 @@ public DataTable execute() {
     long startTime = System.currentTimeMillis();
     InstanceResponseOperator instanceResponseOperator = _instanceResponsePlanNode.run();
     long endTime1 = System.currentTimeMillis();
+    if (endTime1 > _instanceResponsePlanNode._queryContext.getEndTimeMs()) {
+      throw new RuntimeException("Timeout after plan generation");
```

Comment:
Should we throw a TimeoutException?

Reply:
Done

Reply:
Caught and print in [BaseQueriesTest.java](https://github.com/apache/pinot/pull/9386/commits/e17f3492f78a7505f2fa1971b05ea7f8851ffde4#diff-a418235c421d9d09b2328265ddedb7424839f9a6722f8bf4dacacba624f5f1ab) or it will incur too many changes for test cases